pkg/nsx/services/firewall.go

```diff
@@ -161,6 +161,10 @@ func (service *SecurityPolicyService) DeleteSecurityPolicy(obj interface{}) erro
 			log.Error(err, "failed to get security policy", "UID", string(sp))
 			return err
 		}
+		if len(indexResults) == 0 {
+			log.Info("failed to get security policy with index", "UID", string(sp))
```

Comment:
How about `did not get security policy with index`?
And in which case `len(indexResults)` will be 0?

Reply:
ok
It's hard to figure out why there are garbage groups below when running pipelines, but for defensive programming， it's better to add this line.

https://github.com/vmware-tanzu/nsx-operator/blob/1b9f49970085a0893b44c91de81074e8d8c32694/pkg/nsx/services/store.go#L263

![image](https://user-images.githubusercontent.com/4385934/191147835-71d6e1b6-b726-4ffb-bfd3-ebe5ed696fdc.png)

Reply:
ok
I found the bug, because the associated groups are not deleted when deleting the security policy.